src/Processors/Sources/PostgreSQLSource.cpp

```diff
@@ -191,6 +191,12 @@ PostgreSQLSource<T>::~PostgreSQLSource()
     {
         try
         {
+            if (stream)
+            {
+                tx->conn().cancel_query();
```

Comment:
It seems that the transaction destructor calls `abort()`, which cancels the query on the PostgreSQL side. Am I missing something?

Reply:
We use libpqxx `stream_from`, that internally uses `COPY TO STDOUT` query https://www.postgresql.org/docs/current/sql-copy.html, instead of cursors.
When transaction is destroyed it calls `ROLLBACK`, but I tested and until query is finished transaction is stuck. For query that reads all data from table this stuck can take couple of minutes.

Reply:
The change looks at least not harmful to me.
Still, let's add a comment about why this is needed.